db/schema.go

```diff
@@ -171,23 +181,82 @@ func substituteSchemaPatch(
 	patch jsonpatch.Patch,
 	collectionsByName map[string]client.CollectionDescription,
 ) (jsonpatch.Patch, error) {
+	fieldIndexesByName := make(map[string]map[string]int, len(collectionsByName))
+	for colName, col := range collectionsByName {
+		colFieldIndexesByName := make(map[string]int, len(col.Schema.Fields))
+		fieldIndexesByName[colName] = colFieldIndexesByName
+		for i, field := range col.Schema.Fields {
+			colFieldIndexesByName[field.Name] = i
+		}
+	}
```

Comment:
suggestion: This can be simplified to
```go
for colName, col := range collectionsByName {
    fieldIndexesByCollection[colName] = make(map[string]int, len(col.Schema.Fields))
    for i, field := range col.Schema.Fields {
        fieldIndexesByCollection[colName][field.Name] = i
    }
}
```

Reply:
It is one less line, but an extra call.  I prefer the current, there is actually less to read even though it takes up more eye-space.